### PR TITLE
Feat/log details

### PR DESCRIPTION
**Pull request by juskek**
temp

### CODE REVIEW

File: front/src/app/LogsTable.tsx [hunk 1]
```diff
@@ -1,9 +1,13 @@
+"use client";
```
Comment (Lerri-Cofannos) [inline front/src/app/LogsTable.tsx:1]: What is that line for? It is the first time I see this!

File: front/src/app/logs/[id]/error.tsx
Comment (Lerri-Cofannos) [inline front/src/app/logs/[id]/error.tsx:1]: As this is the same component as `app/errors.tsx`, would it not be possible to import and export it?

File: front/src/app/page.tsx
Comment (Lerri-Cofannos) [inline front/src/app/page.tsx:1]: Would you not want to replace the return function with the LogsTable's code? This component became sad all of a sudden

**Review by Lerri-Cofannos — COMMENTED** [review]
The UI looks great!

File: front/src/app/LogsTable.tsx (hunk 1 above)
Comment (juskek) [reply front/src/app/LogsTable.tsx:1]: It's NextJS syntax to state that the component and its children will be bundled to the client

File: front/src/app/page.tsx
Comment (juskek) [reply front/src/app/page.tsx:1]: The code in the LogsTable component is client side, so if I move that code here, I will need to mark this as client side as well - then the LogsData.getLogs() function gets bundled into the client bundle, and exposes the database call

File: front/src/app/logs/[id]/error.tsx
Comment (juskek) [reply front/src/app/logs/[id]/error.tsx:1]: good catch!

File: front/src/app/LogsTable.tsx [hunk 2]
```diff
@@ -28,12 +32,18 @@ export default function LogsTable({ logs }: Props) {
                 </thead>
                 <tbody>
                     {logs.map((log) => (
-                        <tr key={log.id} className="border-b border-gray-200">
+                        <tr
+                            key={log.id}
+                            className="hover:bg-gray-200 cursor-pointer"
+                            onClick={() => router.push(`logs/${log.id}`)}
```
Comment (CallumHemsley) [inline front/src/app/LogsTable.tsx:38]: Is this typed? I wonder if theres a way we can make it typed 🤔 (the routes)

File: front/src/app/error.tsx [hunk 3]
```diff
@@ -10,14 +9,5 @@ export default function Error({
...
-        console.error(error);
-    }, [error]);
-
-    return (
-        <Scaffold>
-            <h2>Something went wrong!</h2>
-            <button onClick={() => reset()}>Try again</button>
-        </Scaffold>
-    );
+    return <GenericError error={error} reset={reset} />;
```
Comment (CallumHemsley) [inline front/src/app/error.tsx:12]: Any reason for this being it's own component if it just returns GenericError?

File: front/src/app/logs/[id]/loading.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export default function Loading() {
+    return (
+        <div className="border bg-white shadow rounded-md p-4 w-full mx-auto">
+            <div role="status" className="max-w-sm animate-pulse">
+                <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-48 mb-4"></div>
```
[the remark this answers is not in the document]
Comment (CallumHemsley) [reply front/src/app/logs/[id]/loading.tsx:5]: Also next is so nice now, the whole `loading` convention with suspense is SICK

[the remark this answers is not in the document]
Comment (CallumHemsley) [reply front/src/app/logs/[id]/loading.tsx:5]: My only question is where is the `Suspense` component? I can't see it 👁️ see here: https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#instant-loading-states

File: front/src/app/logs/[id]/loading.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+                <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 max-w-[360px] mb-2.5"></div>
+                <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 mb-2.5"></div>
+                <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 max-w-[330px] mb-2.5"></div>
+                <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 max-w-[300px] mb-2.5"></div>
+                <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 max-w-[360px]"></div>
+                <span className="sr-only">Loading...</span>
```
Comment (CallumHemsley) [inline front/src/app/logs/[id]/loading.tsx:11]: Personally don't think the loading text is neeeded, if we have a loading skeleton :P

File: front/src/app/logs/[id]/loading.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (CallumHemsley) [reply front/src/app/logs/[id]/loading.tsx:5]: or perhaps with new next features, suspense is directly in if you make it an `async` function?

File: front/src/app/error.tsx (hunk 3 above)
Comment (juskek) [reply front/src/app/error.tsx:12]: There were two error.tsx files just now in app/ and app/logs/[id], but I realised it was unnecessary as all nested routes can redirect to the parent error.tsx

forgot to remove this after the refactor, good catch!

File: front/src/app/logs/[id]/loading.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (juskek) [reply front/src/app/logs/[id]/loading.tsx:5]: yes! it's really intuitive as well :) 

The Suspense is wrapped in automatically when you have a loading file 

image

File: front/src/app/logs/[id]/loading.tsx (hunk 5 above)
Comment (juskek) [reply front/src/app/logs/[id]/loading.tsx:11]: good catch!

File: front/src/app/LogsTable.tsx (hunk 2 above)
Comment (juskek) [reply front/src/app/LogsTable.tsx:38]: done! 
image

**Review by Lerri-Cofannos — APPROVED** [review]
The new atom components are great!

File: front/src/app/LogsTable.tsx (hunk 2 above)
Comment (CallumHemsley) [reply front/src/app/LogsTable.tsx:39]: <3 you love to see it

File: front/next.config.js [hunk 6]
```diff
@@ -1,4 +1,8 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {}
+const nextConfig = {
+    experimental: {
+        typedRoutes: true,
```
Comment (CallumHemsley) [inline front/next.config.js:4]: 🔥

**Pull request merged**